src/contracts/GPv2Settlement.sol

```diff
@@ -191,58 +191,63 @@ contract GPv2Settlement is ReentrancyGuard, StorageAccessible {
     /// [`computeTradeExecution`] for more details.
     /// @param tokens An array of ERC20 tokens to be traded in the settlement.
     /// @param clearingPrices An array of token clearing prices.
-    /// @param encodedTrades Encoded trades for signed orders.
+    /// @param trades Trades for signed orders.
     /// @return executedTrades Array of executed trades.
     function computeTradeExecutions(
         IERC20[] calldata tokens,
         uint256[] calldata clearingPrices,
-        bytes calldata encodedTrades
+        GPv2Trade.Data[] calldata trades
     ) internal returns (GPv2TradeExecution.Data[] memory executedTrades) {
-        (uint256 tradeCount, bytes calldata remainingEncodedTrades) =
-            encodedTrades.decodeTradeCount();
-        executedTrades = new GPv2TradeExecution.Data[](tradeCount);
-
-        GPv2Encoding.Trade memory trade;
-        uint256 i = 0;
-        while (remainingEncodedTrades.length != 0) {
-            remainingEncodedTrades = remainingEncodedTrades.decodeTrade(
+        GPv2Signing.RecoveredOrder memory recoveredOrder =
```

Comment:
nanonit: at first I didn't get why we allocate here instead of in `recoverOrderFromTrade`. I assume for you the gas efficiency argument is like

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2FBoC8nVlu3oNRsBJw30%2Fgiphy.gif&token=duh" data-gitmeme-token="duh"><img src="https://media3.giphy.com/media/BoC8nVlu3oNRsBJw30/giphy.gif" title="Created by gitme.me with /duh"/></a>

Reply:
This is to reuse the `GPv2Signing.RecoveredOrder memory` storage for recovering an order's owner.

Reply:
I'm going to quantify this and get back to you.

Reply:
So it turns out the cost isn't as high as it once once (since the memory we return has shrunk):
```
           10 |          100 |           10 |           20 |      7411147  (change: 637 / trade)
```

Reply:
Yes, I understand now. Maybe a comment would have been useful for the non-gas aware reader.